cmd/bucket-replication.go

```diff
@@ -3148,19 +3173,23 @@ func (p *ReplicationPool) persistMRF() {
 			if !ok {
 				return
 			}
-			if len(entries) >= mrfMaxEntries {
+			if len(p.mrfEntries) >= mrfMaxEntries {
 				saveMRFToDisk()
 			}
-			entries[e.versionID] = e
+			p.mrfEntries[e.versionID] = e
 		}
 	}
 }
 
-func (p *ReplicationPool) queueMRFSave(entry MRFReplicateEntry) {
+// queueMRFSave queues most recent failures to disk if number of retries not exceed limit and if target
+// is not offline for more than 10 minutes.
+func (p *ReplicationPool) queueMRFSave(entry MRFReplicateEntry, ep string) {
 	if !p.initialized() {
 		return
 	}
-	if entry.RetryCount > mrfRetryLimit { // let scanner catch up if retry count exceeded
+	// drop if retry count exceeded or target is offline for more than 10 minutes and mrf map is full. Note that we may
+	// not always know the target endpoint if failure is early on in the replication flow
+	if entry.RetryCount > mrfRetryLimit || (ep != "" && globalBucketTargetSys.offlineFor(ep) > 10*time.Minute && len(p.mrfEntries) >= mrfMaxEntries) { // let scanner catch up if retry count exceeded
```

Comment:
```suggestion
	if entry.RetryCount > mrfRetryLimit || (len(p.mrfEntries) >= mrfMaxEntries && globalBucketTargetSys.offlineSince(ep, 10*time.Minute)) { // let scanner catch up if retry count exceeded
```

This is an optimized check with `&&` operator if the first entry says `false` Golang optimizes and avoids calling the second operand.